components/molecules/SuggestedRepo/suggested-repo.tsx

```diff
@@ -4,12 +4,22 @@ import RepoCardProfile, { RepoCardProfileProps } from "../RepoCardProfile/repo-c
 interface SuggestedRopsitoryProps {
   data: RepoCardProfileProps;
   onAddRepo?: (repo: string) => void;
+  loadingData?: {
+    repoName: string;
+    isLoading: boolean;
+    isAddedFromCart: boolean;
+  };
 }
-const SuggestedRepository = ({ data, onAddRepo }: SuggestedRopsitoryProps) => {
+const SuggestedRepository = ({ data, onAddRepo, loadingData }: SuggestedRopsitoryProps) => {
   return (
     <div className="flex items-center justify-between">
       <RepoCardProfile {...data} />
       <Button
+        disabled={
+          `${data.orgName}/${data.repoName}` === loadingData?.repoName &&
+          loadingData?.isLoading &&
+          !loadingData?.isAddedFromCart
+        }
```

Comment:
```suggestion
        disabled={
        repos.length === insightRepoLimit ||
          (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
        }
        loading={
        repos.length === insightRepoLimit ||
          (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
        }
```

Reply:
Thank you for the thoughtful suggestion!

Reply:
Thank you for the thoughtful suggestion!